[MIG][9.0] base_export_manager

lasley opened the pull request:
This PR migrates base_export_manager to v9

No changes were required outside of the standard upgrades. I did notice a remnants of the old naming in the JS, which I resolved while v9-ifying it

pedrobaeza committed:
[REN] base_exports_manager > base_export_manager

pedrobaeza committed:
Update README.rst

oca-transbot committed:
OCA Transbot updated translations from Transifex

hhgabelgaard committed:
Missing rename

pedrobaeza committed:
Merge pull request #308 from steingabelgaard/8.0-export-hhg

Missing rename?

yajo committed:
Improve UX for base_export_manager.

- Improve user instructions in README.
- Require some required fields.
- Allow to select models from a list.
- Allow to select up to 3 fields from dynamic lists.
- Improve translations.
- More tests.
- Translate column labels.

Some methods have been renamed, so version tag is raised to 8.0.2.0.0.

yajo committed:
Fix test.

yajo committed:
Make inverse method be called at view time.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

lasley committed:
[MIG] base_export_manager: Upgrade to v9
* Upgrade base_export_manager to v9
* Rename JS to base_export_manager

rafaelbn commented:
@lasley thanks! but we mus wait for this module... it's still pending an improvement in v8 very important https://github.com/OCA/server-tools/pull/317

lasley commented:
Thanks for the pointer towards the open 8.0 PR @rafaelbn - seems I missed that. I will review that PR sometime today & bring this way when merged.

yajo committed:
README & author updates.

yajo committed:
Hardcode constraints in the `create` method instead of using normal ones.

Depending on the context where the record is created, it gets `resource` or `model_id`. The problem is that Odoo checks constrains before inverses, so constrains would fail always.

Test added to ensure future versions contemplate all use cases.

yajo committed:
Allow to reorder fields.

rafaelbn commented:
Hi @lasley , https://github.com/OCA/server-tools/pull/317 is ready to be merged after that we can continue with this one.

rafaelbn commented:
Hi @lasley , https://github.com/OCA/server-tools/pull/317 is merged. Could you bring please all to this PR in order to test it? Thanks!

lasley committed:
Merge branch '8.0-extract' into hotpatch/9.0/base_export_manager

lasley commented:
Rebased

pedrobaeza commented on base_export_manager/models/ir_exports.py:
```diff
@@ -0,0 +1,67 @@
+        "ir.model",
+        "Model",
+        store=True,
+        domain=[("osv_memory", "=", False)],
```
This domain is no longer valid in v9

lasley committed:
[REF] base_export_manager: Fix model domain
* Change osv_memory to transient for model_id domain

lasley commented:
Thanks for the catch @pedrobaeza - fixed!

rafaelbn commented:
Super! Tested in runbot! :+1:

max3903 commented:
:+1:

pedrobaeza commented:
:+1:
